src/main/java/org/eclipse/golo/runtime/ArrayHelper.java

```diff
@@ -32,6 +32,13 @@ public static Object head(Object[] array) {
     return new Object[0];
   }
 
+  public static Object last(Object[] array) {
```

Comment:
I'm not sure. Since it's a partial method, we can either:
- throw an exception (`NoSuchElementException` to be consistent with the `LinkedList::getLast` method that can be called as `last` in golo, or even `NoSuchMethodException` to effectively limit the method to non-empty arrays, but since it's a runtime check, it does not really matter),
- return an `Optional` (semantically nice, but more FP oriented),
- return `null` (with known issues)

Given the elvis operator (`?:`), `null` is somewhat a OOP equivalent of the FP `Option`. In this case, returning `null` allows to `myarray: last()?: someMethod() orIfNull "foo"`. It's also consistent with `head/first` that returns `null` and `tail` that returns an empty array instead of throwing an exception.
Moreover, `null` can lead to `NullPointerException`, but raising a `NoSuchElementException` lead to the same kind of behavior (either test for emptiness or wrap in `try/catch`); however, the exception can raise anywhere (when the value is used), contrary to when the method is called
On the other hand, returning `null` preclude the use of lists and arrays in a consistent way, unless we also augment lists with a `first/last` method shadowing the java `getFirst/getLast` ones.